packages/host/app/components/operator-mode/create-file-modal.gts

```diff
@@ -87,4 +153,58 @@ export default class CreateFileModal extends Component<Signature> {
   @action removeSelectedRealm() {
     this.selectedRealmURL = undefined;
   }
+
+  private chooseCardInstanceType = restartableTask(async () => {
+    this.selectedCatalogEntry = await chooseCard({
+      filter: {
+        on: catalogEntryRef,
+        eq: { isField: false },
+      },
+    });
+  });
+
+  private createCardInstance = restartableTask(async () => {
+    if (!this.selectedCatalogEntry?.ref || !this.selectedRealmURL) {
+      return;
+    }
+
+    let { ref } = this.selectedCatalogEntry;
+
+    let doc: LooseSingleCardDocument = {
+      data: {
+        meta: {
+          adoptsFrom: ref,
+          realmURL: this.selectedRealmURL.href,
+        },
+      },
+    };
+
+    let relativeTo = new URL(this.selectedCatalogEntry.id);
+    let cardModule = new URL(moduleFrom(ref), relativeTo);
+    // we make the code ref use an absolute URL for safety in
+    // the case it's being created in a different realm than where the card
+    // definition comes from
+    if (!new RealmPaths(this.selectedRealmURL).inRealm(cardModule)) {
+      let maybeRef = codeRefWithAbsoluteURL(ref, relativeTo);
+      if ('name' in maybeRef && 'module' in maybeRef) {
+        ref = maybeRef;
+      }
+    }
+
+    let card = await this.cardService.createFromSerialized(
+      doc.data,
+      doc,
+      relativeTo,
+    );
+
+    if (card) {
+      let savedCard = await this.cardService.saveModel(this, card);
+      console.log(savedCard?.id);
+      if (savedCard) {
```

Comment:
I believe `this.cardService.saveModel()` actually just mutates the existing model that you pass in. probably this guard is unnecessary and you can just use `card` below. there is a reason that this does have a return--i think types reveal that reason if you make this method have no return...